internal/testrunner/runners/system/runner.go

```diff
@@ -912,3 +944,45 @@ func (r *runner) generateTestResult(docs []common.MapStr) error {
 
 	return nil
 }
+
+func (r *runner) anyErrorMessages(profile *profile.Profile, serviceName string) error {
+	if _, found := errorPatterns[serviceName]; !found {
+		logger.Debugf("No error patterns defined for %s", serviceName)
+		return nil
+	}
+
+	tempDir, err := os.MkdirTemp("", "test-system-")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDir)
+
+	logs, err := stack.ParseLogs(stack.ParseLogsOptions{
+		ServiceName: serviceName,
+		Profile:     r.options.Profile,
+		LogsPath:    tempDir,
+	})
+	if err != nil {
+		return err
+	}
+
+	var multiErr multierror.Error
+
+	for _, pattern := range errorPatterns[serviceName] {
+		r := regexp.MustCompile(pattern)
+
+		for _, log := range logs {
```

Comment:
Is it possible that the logs may be poluted by messages of initialization or previous tests? Should we take into account the timestamps, to select logs happened during the execution of the test?

Reply:
That's true, it could happen if the stack is kept between different runs of tests.
Added logic to just process the logs after the starting time of the system test execution.